src/renderer/contextMenu/tabs/index.js

```diff
@@ -0,0 +1,32 @@
+import { remote } from 'electron'
+import {
+  CLOSE_THIS,
+  CLOSE_OTHERS,
+  CLOSE_SAVED,
+  CLOSE_ALL,
+  SEPARATOR,
+  RENAME,
+  COPY_PATH,
+  SHOW_IN_FOLDER
+} from './menuItems'
+
+const { Menu, MenuItem } = remote
+
+export const showContextMenu = (event, tab) => {
+  const menu = new Menu()
+  const win = remote.getCurrentWindow()
+  const { pathname } = tab
+  const CONTEXT_ITEMS = [CLOSE_THIS, CLOSE_OTHERS, CLOSE_SAVED, CLOSE_ALL, SEPARATOR, RENAME, COPY_PATH, SHOW_IN_FOLDER]
+  const FILE_CONTEXT_ITEMS = [RENAME, COPY_PATH, SHOW_IN_FOLDER]
+
+  FILE_CONTEXT_ITEMS.forEach(item => {
+    item.enabled = !!pathname
+  })
+
+  CONTEXT_ITEMS.forEach(item => {
+    const menuItem = new MenuItem(item)
+    menuItem.tab = tab
```

Comment:
This is not the best way to solve the problem but please only use tab id (`tab.id`). You can get the original tab by id in editor. Maybe a function call with the tab id to construct the item would be a better way.

Reply:
How about menuItem._tabId ? add a prefix `_`